lib/screens/login/login_repository.dart

```diff
@@ -12,14 +15,50 @@ class FakeLoginRepository implements LoginRepository {
   @override
   Future<String> login(String email, String password) async {
     // Simulate network delay
+    SharedPreferences sharedPreferences = await SharedPreferences.getInstance();
+
     await Future.delayed(Duration(seconds: 1));
 
     if (Random().nextBool()) {
       // random network error
       throw NetworkException();
     } else {
       // fake successful login
-      return "Token abcdEfgh";
+      sharedPreferences.setString(S.auth, "Token 9935a8b04f2de7f5dec8f9e92a1893822b034dc7");
+      return "Token 9935a8b04f2de7f5dec8f9e92a1893822b034dc7";
+    }
+  }
+}
+
+class APILoginRepository implements LoginRepository {
+  @override
+  Future<String> login(String email, String password) async {
+    http.Response response;
+    SharedPreferences sharedPreferences = await SharedPreferences.getInstance();
+    try {
+      response = await http.post(
+        S.loginUrl,
+        body: <String, dynamic>{S.emailKey: email, S.passwordKey: password},
+      );
+      sharedPreferences.setString(S.auth, json.encode(response));
```

Comment:
Wrong according to bloc: This is not to be done here. It is to be done in cubit (explained above) (**fix this**)
Semantically wrong: you have (**this is for understanding purposes**)
1. You are attempting to encode the response, where it is supposed to be decoded. Read more about Encoding and Decoding.
2. You have saved the entire response in shared preference, when you have to only save the token.